views/reducers/session.jsx

```diff
@@ -15,13 +16,22 @@ const user = function user(state = {
         isProcessing: false,
         signedIn: true,
         email: action.email,
+        authError: false,
       });
     case SIGNED_OUT:
       return Object.assign({}, state, {
         isProcessing: false,
         signedIn: false,
         email: '',
       });
+    case AUTH_ERROR:
+      return Object.assign({}, state, {
+        authError: true,
```

Comment:
I think we should also add
        isProcessing: false,
because technically we are done processing the user and we know the results which is failure. For some reason, it is getting reset implicitly. It could be the same issue with onLeave.